crates/pathfinder/src/sync/p2p.rs

```diff
@@ -435,6 +442,53 @@ impl Sync {
 
         Ok(())
     }
+
+    async fn sync_class_definitions(&self, stop: BlockNumber) -> anyhow::Result<()> {
+        if let Some(start) = class_definitions::next_missing(self.storage.clone(), stop)
+            .await
+            .context("Finding next block with missing class definition(s)")?
+        {
+            let result = self
+                .p2p
+                .clone()
+                .class_definitions_stream(
+                    start,
+                    stop,
+                    class_definitions::declared_class_counts_stream(
+                        self.storage.clone(),
+                        start,
+                        stop,
+                    ),
+                )
+                .map_err(Into::into)
+                .and_then(class_definitions::verify_layout)
+                .and_then(class_definitions::verify_hash)
+                .try_chunks(1000)
```

Comment:
What's the maximum class definition size? This might be a bit much.

Some safer options would be a manual implementation that buffers up to some maximum accumulated byte size?

But maybe for now we just leave this out.

Reply:
Yeah, 1000 classes is too much, will fix it.